allennlp/modules/text_field_embedders/model_text_field_embedder.py

```diff
@@ -0,0 +1,72 @@
+from typing import Dict
+
+import torch
+from overrides import overrides
+
+from allennlp.common import Params
+from allennlp.data import Vocabulary
+from allennlp.models.model import Model
+from allennlp.modules.text_field_embedders.text_field_embedder import TextFieldEmbedder
+
+
+@TextFieldEmbedder.register("model")
+class ModelTextFieldEmbedder(TextFieldEmbedder):
+    """
+    This is a ``TextFieldEmbedder`` that wraps a ``Model``.  The model must accept a single text
+    field as input.  We run ``forward`` on the model, then pull out one of the values in its output
+    dictionary to return.
+
+    NOTE: This is still largely experimental and needs some real-world testing before I'm confident
+    this API is correct and actually works well.
+
+    Parameters
+    ----------
+    model : ``Model``
+        The model that we will use to embed the text field.
+    model_input_name : ``str``
```

Comment:
This won't work in the general case, I think (see my note about the SRL model below) - in order to run the model forward, it must have everything required apart from the `labels` etc. Instead, we should just have `forward` take `kwargs` and unpack them into `forward` of the wrapped model.